lib/netext/httpext/transport.go

```diff
@@ -243,6 +243,9 @@ func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	reqWithTracer := req.WithContext(httptrace.WithClientTrace(ctx, tracer.Trace()))
 	resp, err := t.state.Transport.RoundTrip(reqWithTracer)
 
+	if typErr, ok := err.(net.Error); ok && typErr.Timeout() {
+		err = NewK6Error(requestTimeoutErrorCode, requestTimeoutErrorCodeMsg, err)
+	}
```

Comment:
I might be misunderstanding here but I think the `err` type returned from `RoundTrip` is a `context.deadlineExceededError`. This means that it'll never end up making it into `errorCodeForNetOpError` [here](https://github.com/k6io/k6/blob/baa5a8af1591ef3c8aa5d5cb1540bceb4a550b6e/lib/netext/httpext/error_codes.go#L204). This is also shown by the fact that `context deadline exceeded` errors return a code of 1000 which is not returned anywhere from `errorCodeForNetOpError`.

I think that given that these errors will be completely separate times, it should prevent us from masking the `dial` related errors.

Reply:
The error you create here and then get set in the struct below is the one that then goes through the error_code matching code and `net.OpError` (the one that is matched in there) does implement `net.Error` so it will definitely mask it ;).

On the other hand, I am not really certain it is needed, the original code (written by me) was just me trying to match as many errors as we have seen "somehow" which was over a year ago and so the current niceties in the `errors` package weren't there. 

I probably should've had just one timeout error as it is here and had it do everything. 

I am not even certain we have had `dial timeout error` at any point in time or have I just by reading the `net` code found that this will match something "hopefully" and added it, I certainly tried this for a lot of errors that turned out later weren't matched correctly, because I didn't get the exact wrapping correct and there wasn't [errors.As](https://golang.org/pkg/errors/#As) :(.

So I am fine with just dropping the dial timeout error code (and code) and having the rest of the change as it is.

Reply:
Ok - if we think that creating a generic "timeout" error code is more useful then I've removed the dial conditional of the netOp error branch. Let me know if this isn't what you had in mind.